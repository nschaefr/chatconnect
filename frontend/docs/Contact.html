<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width,initial-scale=1" />
    <title>Contact - Documentation</title>

    <script src="scripts/prettify/prettify.js"></script>
    <script src="scripts/prettify/lang-css.js"></script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link
      type="text/css"
      rel="stylesheet"
      href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css"
    />
    <link
      type="text/css"
      rel="stylesheet"
      href="styles/prettify-tomorrow.css"
    />
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css" />
  </head>
  <body>
    <input type="checkbox" id="nav-trigger" class="nav-trigger" />
    <label for="nav-trigger" class="navicon-button x">
      <div class="navicon"></div>
    </label>

    <label for="nav-trigger" class="overlay"></label>

    <nav>
      <li class="nav-link nav-home-link"><a href="index.html">Home</a></li>
      <li class="nav-heading">Classes</li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Avatar.html">Avatar</a></span>
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Avatar.html#~getColorByHash">getColorByHash</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Avatar.html#~hashCode">hashCode</a></span
        >
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Chat.html">Chat</a></span>
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~connectToWebSocket">connectToWebSocket</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~filterbyKeyword">filterbyKeyword</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~handleMessage">handleMessage</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~handleResize">handleResize</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~logout">logout</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~sendMessage">sendMessage</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="Chat.html#~showOnlinePeople">showOnlinePeople</a></span
        >
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Contact.html">Contact</a></span>
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"><a href="Routes.html">Routes</a></span>
      </li>
      <li class="nav-heading">
        <span class="nav-item-type type-class">C</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html">SignuploginForm</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~submitHandler">submitHandler</a></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~validatePassword"
            >validatePassword</a
          ></span
        >
      </li>
      <li class="nav-item">
        <span class="nav-item-type type-function">F</span
        ><span class="nav-item-name"
          ><a href="SignuploginForm.html#~validateUsername"
            >validateUsername</a
          ></span
        >
      </li>
    </nav>

    <div id="main">
      <h1 class="page-title">Contact</h1>

      <section>
        <header>
          <h2>Contact</h2>
        </header>

        <article>
          <div class="container-overview">
            <div class="section-method">
              <h4 class="name" id="Contact">
                <span class="type-signature"></span>new Contact<span
                  class="signature"
                  >(id, username, onClick, selected, screen)</span
                ><span class="type-signature"></span>
              </h4>

              <div class="description">
                <p>Component which shows the contact list</p>
              </div>

              <dl class="details">
                <dt class="tag-source">Source:</dt>
                <dd class="tag-source">
                  <ul class="dummy">
                    <li>
                      <a href="components_chat-page_contact.jsx.html"
                        >components/chat-page/contact.jsx</a
                      >,
                      <a href="components_chat-page_contact.jsx.html#line12"
                        >line 12</a
                      >
                    </li>
                  </ul>
                </dd>
              </dl>

              <h5>Parameters:</h5>

              <table class="params">
                <thead>
                  <tr>
                    <th>Name</th>

                    <th>Type</th>

                    <th class="last">Description</th>
                  </tr>
                </thead>

                <tbody>
                  <tr>
                    <td class="name"><code>id</code></td>

                    <td class="type">
                      <span class="param-type"><code>string</code></span>
                    </td>

                    <td class="description last">
                      <p>User id</p>
                    </td>
                  </tr>

                  <tr>
                    <td class="name"><code>username</code></td>

                    <td class="type">
                      <span class="param-type"><code>string</code></span>
                    </td>

                    <td class="description last">
                      <p>Username</p>
                    </td>
                  </tr>

                  <tr>
                    <td class="name"><code>onClick</code></td>

                    <td class="type">
                      <span class="param-type"><code>function</code></span>
                    </td>

                    <td class="description last">
                      <p>onClick function to set selected user id</p>
                    </td>
                  </tr>

                  <tr>
                    <td class="name"><code>selected</code></td>

                    <td class="type">
                      <span class="param-type"><code>string</code></span>
                    </td>

                    <td class="description last">
                      <p>Selected user</p>
                    </td>
                  </tr>

                  <tr>
                    <td class="name"><code>screen</code></td>

                    <td class="type">
                      <span class="param-type"><code>string</code></span>
                    </td>

                    <td class="description last">
                      <p>Mobile or browser view</p>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </article>
      </section>
    </div>

    <br class="clear" />

    <footer>
      Generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 4.0.2</a> on
      Thu Sep 07 2023 18:27:20 GMT+0200 (Central European Summer Time) using the
      Minami theme.
    </footer>

    <script>
      prettyPrint();
    </script>
    <script src="scripts/linenumber.js"></script>
  </body>
</html>
