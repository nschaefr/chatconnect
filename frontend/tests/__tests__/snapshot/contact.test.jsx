import React from "react";
import { render } from "@testing-library/react";
import Contact from "../../../src/components/chat-page/contact";

describe("contact snapshots", () => {
  test("render contact component", () => {
    const { container } = render(<Contact id="123" username={"testuser"} />);

    expect(container).toMatchSnapshot();
  });
});
