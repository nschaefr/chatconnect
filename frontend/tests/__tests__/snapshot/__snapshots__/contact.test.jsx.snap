// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`contact snapshots render contact component 1`] = `
<div>
  <div
    class="contactDiv"
    data-testid="contact"
    style="padding: 10px 0px 10px 20px; display: flex; align-items: center; color: white; gap: 10px; margin-left: 15px; margin-right: 15px; cursor: pointer; border-radius: 3px; background-color: rgb(40, 40, 60);"
  >
    <div
      data-testid="avatar"
      style="background-color: rgb(153, 207, 201); width: 35px; height: 35px; display: flex; border-radius: 50%; align-items: center; justify-content: center;"
    >
      <div
        style="font-family: ExtraBold; color: white; font-size: 15px;"
      >
        T
      </div>
    </div>
    <div
      class="contactNameDiv"
    >
      <span
        style="font-size: 15px; font-family: SemiBold;"
      >
        testuser
      </span>
    </div>
  </div>
</div>
`;
